test/buy-tshirt.spec.ts

```diff
@@ -1,38 +1,55 @@
 import { $, browser } from 'protractor';
+import { MenuContentPage, AddressStepPage, BankPaymentPage, ProductAddedPage,
+  ProductListPage, SummaryContentPage, SignInContent, OrderSummaryPage,
+  ShippingStepPage, PaymentStepPage } from '../src/page';
 
 describe('Buy a t-shirt', () => {
+  const menuContentPage: MenuContentPage = new MenuContentPage();
+  const addressContentPage: AddressStepPage = new AddressStepPage();
+  const bankContentPage: BankPaymentPage = new BankPaymentPage();
+  const productAddedPage: ProductAddedPage = new ProductAddedPage();
+  const productListPage: ProductListPage = new ProductListPage();
+  const summaryContentPage: SummaryContentPage = new SummaryContentPage();
+  const singInContentPage: SignInContent = new SignInContent();
+  const orderSummaryPage: OrderSummaryPage = new OrderSummaryPage();
+  const shippingStepPage: ShippingStepPage = new ShippingStepPage();
+  const paymentStepPage: PaymentStepPage = new PaymentStepPage();
+
   beforeEach(() => {
     jasmine.DEFAULT_TIMEOUT_INTERVAL = 120000;
   });
 
   it('then should be bought a t-shirt', async () => {
     await browser.get('http://automationpractice.com/');
     await(browser.sleep(10000));
-    await $('#block_top_menu > ul > li:nth-child(3) > a').click();
+    await menuContentPage.goToTShirtMenu();
     await(browser.sleep(3000));
-    await $('#center_column a.button.ajax_add_to_cart_button.btn.btn-default').click();
+    // product-added-modal.page
+    await productAddedPage.addProduct();
```

Comment:
I guess you got confused, The modal page is the next step, and the productList page is this step :)